File: test-fixtures/sample-repo/csharp/UserService.cs
using Apex.Data;

namespace Apex.Service;

public interface IUserReader
{
    string? FindUser(string id);
}

public class UserService : IUserReader
{
    private readonly UserRepository repository;

    public UserService(UserRepository repository)
    {
        this.repository = repository;
    }

    public string? FindUser(string id)
    {
        return repository.Find(id);
    }
}

File: test-fixtures/sample-repo/java/UserService.java
import org.springframework.stereotype.Service;

@Service
public class UserService implements IUserService {
    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public User findUser(String id) {
        return userRepository.find(id);
    }
}

